### PR TITLE
Fix ZstdFilter ignoring the requested compression level

### PR DESCRIPTION
**Pull request by giordano**
apply_filter! passed the level to ZstdEncodeOptions as the keyword `level`, but ChunkCodecLibZstd names it `compressionLevel` and its constructor silently discards unknown keywords, so every ZstdFilter compressed at the default level regardless of the requested one.

The options construction is factored out into a new `Filters.encode_options(filter)` function, so the parameter plumbing is directly testable: the regression test checks that the options constructed for ZstdFilter and Deflate carry the requested level, and that non-default levels round-trip through the file API.

### CODE REVIEW

File: src/Filters.jl [hunk 1]
```diff
@@ -41,6 +41,14 @@ Depending on the filter type, this may include something like a compression leve
 """
 client_values(::Filter) = ()
 
+"""
+    encode_options(filter::Filter)
+
+Construct the `ChunkCodecCore.EncodeOptions` used to encode data with `filter`,
+carrying over the filter's parameters (such as the compression level).
+"""
+function encode_options end
```
Comment (giordano) [inline src/Filters.jl]: This function is a bit silly, but it's only for making testing a little bit easier.

File: test/compression.jl [hunk 2]
```diff
@@ -103,6 +103,30 @@ end
...
 end
 
+@testset "Compression level is applied" begin
+    # Regression test: ZstdEncodeOptions names its keyword `compressionLevel`
+    # and silently discards unknown keywords, so passing the level as `level`
+    # made every ZstdFilter compress at the default level regardless of the
+    # requested one. Check that the encode options constructed for a filter
+    # actually carry the requested level.
+    F = JLD2.Filters
+    for level in (1, 3, 22)
```
Comment (nhz2) [inline test/compression.jl]:
```suggestion
    for level in (-100, 1, 3, 22)
```
to also test negative levels

File: src/Filters.jl (hunk 1 above)
Comment (nhz2) [reply src/Filters.jl]: I think this function should be move to line 260 so it is more clear it is a helper function and not part of the `Filter` interface.
